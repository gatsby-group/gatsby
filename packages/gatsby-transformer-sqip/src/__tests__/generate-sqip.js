const { resolve } = require(`path`)

const { remove } = require(`fs-extra`)

const generateSqip = require(`../generate-sqip.js`)

describe(`gatsby-transformer-sqip`, async () => {
  const absolutePath = resolve(
    __dirname,
    `images`,
    `alisa-anton-166247-unsplash-400px.jpg`
  )
  const cacheDir = __dirname

  // const file = getFileObject(absolutePath)

  describe(`generateSqip`, () => {
    it(`generates sqip`, async () => {
      const cache = {
        get: jest.fn(),
        set: jest.fn(),
      }
      const numberOfPrimitives = 5
      const blur = 0
      const mode = 3
      const result = await generateSqip({
        cache,
        cacheDir,
        absolutePath,
        numberOfPrimitives,
        blur,
        mode,
      })
      // @todo snapshots won't help here since primitive changes it's output every time. Consider counting primitives and checking blur setting via cheerio
      expect(result).toMatchSnapshot()
      // @todo delete properly
      await remove(resolve(cacheDir, `alisa-anton-166247-unsplash-400px.svg`))
    })
  })
})
