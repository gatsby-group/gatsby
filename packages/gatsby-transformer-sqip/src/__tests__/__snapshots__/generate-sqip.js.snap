// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-transformer-sqip generateSqip generates sqip 1`] = `
Object {
  "dataURI": "data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 400 200'%3e%3cpath fill='%23514946' d='M0 0h400v200H0z'/%3e%3cg transform='translate(.8 .8) scale(1.5625)' fill-opacity='.5'%3e%3cellipse fill='%23dcf4ff' cx='165' cy='77' rx='34' ry='24'/%3e%3cellipse fill='%23b18770' cx='103' cy='114' rx='49' ry='21'/%3e%3cellipse fill='%23110f18' cx='77' cy='52' rx='34' ry='42'/%3e%3cellipse fill='%23292019' cx='248' cy='23' rx='50' ry='135'/%3e%3cellipse fill='%23c5b3ad' cx='158' cy='60' rx='24' ry='23'/%3e%3c/g%3e%3c/svg%3e",
  "svg": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 400 200\\"><path fill=\\"#514946\\" d=\\"M0 0h400v200H0z\\"/><g transform=\\"translate(.8 .8) scale(1.5625)\\" fill-opacity=\\".5\\"><ellipse fill=\\"#dcf4ff\\" cx=\\"165\\" cy=\\"77\\" rx=\\"34\\" ry=\\"24\\"/><ellipse fill=\\"#b18770\\" cx=\\"103\\" cy=\\"114\\" rx=\\"49\\" ry=\\"21\\"/><ellipse fill=\\"#110f18\\" cx=\\"77\\" cy=\\"52\\" rx=\\"34\\" ry=\\"42\\"/><ellipse fill=\\"#292019\\" cx=\\"248\\" cy=\\"23\\" rx=\\"50\\" ry=\\"135\\"/><ellipse fill=\\"#c5b3ad\\" cx=\\"158\\" cy=\\"60\\" rx=\\"24\\" ry=\\"23\\"/></g></svg>",
}
`;
